Issue#6/relation spot id et residence id sur visit schedule

nfanorenana opened the pull request.

nfanorenana committed:
add relation spot_id and residence_id to visit_schedules

nfanorenana committed:
Add error message

jeni-sowell commented on app/models/visit_schedule.rb:
```diff
@@ -2,8 +2,10 @@ class VisitSchedule < ApplicationRecord
   belongs_to :checklist
+  belongs_to :place
+  belongs_to :residence
+  belongs_to :spot
```
All of this is optional !!!!!!!

jeni-sowell reviewed (commented):
où sont les autres tests ??

jeni-sowell commented on app/models/concerns/visit_schedule_validatable.rb:
```diff
@@ -14,9 +15,44 @@ def cant_set_invalid_due_at
+    depth_level = checklist.location_type.base_location_type.depth_level
+
+    case depth_level
+    when 'place', 'residence', 'spot'
```
Il est préférable de faire un when par étape et non avec virgule .... , déjà est ce que ça marche ?

jeni-sowell commented on app/models/concerns/visit_schedule_validatable.rb:
```diff
@@ -14,9 +15,44 @@ def cant_set_invalid_due_at
+    end
+  end
+
+  def check_compatibility_between_location_and_checklist(location)
```
j'aime bien ce bloque de code

jeni-sowell commented on app/models/visit_schedule.rb:
```diff
@@ -2,8 +2,10 @@ class VisitSchedule < ApplicationRecord
 
-  belongs_to :place
   belongs_to :checklist
+  belongs_to :place
```
hey , souviens toi c'est soit une place soit une residence soit un spot tu trouve normal ce code ?

nfanorenana committed:
update visit_schedule_validatable

- Modification validation cant_set_incompatible_location_and_depth_level
- Modification validation check_compatibility_between_location_and_checklist
- Ajout optional place, residence, spot

nfanorenana committed:
Update issue_type_spec.rb

Creation variable pour le test de validation issue_type

nfanorenana committed:
Update error in visit_schedule validation
